move example of spring web to guide

sdelamo opened the pull request:
To generate the sample projects and the HTML

`./gradlew micronautSpringWebBuild`

You can open the HTML at:

`open build/dist/index.html`

To run the test for every generated project:

```
cd build/code/micronaut-spring-web
./test.sh
```

sdelamo committed:
move example of spring web to guide

To generate the sample projects and the HTML

`./gradlew micronautSpringWebBuild`

You can open the HTML at:

`open build/dist/index.html`

To run the test for every generated project:

```
cd build/code/micronaut-spring-web
./test.sh
```

sdelamo commented:
@mattmoss   To give you a head start, I created the guide skeleton: 

- A new folder under guides with the guide slug `micronaut-spring-web`. That will be part of the guide URL. Thus, it must contain SEO-related terms. We can change this once we understand well what the app is doing. 
- Inside that folder, I created the `metadata.json` file. It defines whether the guide supports multiple languages, the features used by the generated applications, etc. 
- I moved the [deleted code](https://github.com/micronaut-projects/micronaut-graphql/pull/242) in the example project to the guide. 
- I change the sources package to `example.micronaut`. Guides generate sample applications with that package. Moreover, some macros (e.g. `source:`) use that package by default. 
- I created the asciidoc file with the typical sections and added macros to show you how to embed sources or resources. 

Next steps: 

- [ ] The example application have few and failing tests. We have to add tests. We use guides as a functional test suite for Micronaut. Moreover, I think it is important that we show users that they should test their Micronaut applications and how easy and fast is to do that. 
- [ ] Go through every source file and write them as idiomatic as possible. 
- [ ] Write the guide. Add callouts and instructions to Javadoc explaining the sources. 

Let me know if you have any doubts.

sdelamo committed:
Merge branch 'main' into spring-web

sdelamo committed:
guide

sdelamo commented:
I have not included the GraalVM section because it fails to serialize. I will create an issue with instructions about how to reproduce once the guide is published.

timyates commented:
@sdelamo I assume you meant Spring Boot, not Spring Web as in the PR title and description?

timyates committed:
Tweaks

- Add configuration to guide (the GreetingControllerTest fails without it)
- AtomicReference in the Service can be final
- Test classes do not need to be public

timyates reviewed (approved):
I pushed a couple of tweaks in https://github.com/micronaut-projects/micronaut-guides/pull/839/commits/67bb5c9ec1b0acbaaa316397c9b6530f5a823410

I also added the configuration to the guide, as it is required that the response is Spanish for the test to pass 🇪🇸👍

@sdelamo Please take a look 😀

sdelamo commented:
> @sdelamo I assume you meant Spring Boot, not Spring Web as in the PR title and description?

Do you think we should rename the slug from `micronaut-spring-web`to `micronaut-spring-boot`?

timyates commented:
It already is...

Isn't it?

sdelamo commented:
> It already is...
> 
> Isn't it?

It is. 😙

kroegerk commented on guides/micronaut-spring-boot/micronaut-spring-boot.adoc:
```diff
@@ -0,0 +1,169 @@
+
+source:Application[]
+
+<1> In a Spring Boot Application, the https://docs.spring.io/spring-boot/docs/2.0.x/reference/html/using-boot-using-springbootapplication-annotation.html#using-boot-using-springbootapplication-annotation[@SpringBootApplication] enables Spring Boot’s auto-configuration mechanism, enables `@Component` scan on the package where the application is located, and allows to register extra beans in the co ...
```
```suggestion
<1> In a Spring Boot application, the https://docs.spring.io/spring-boot/docs/2.0.x/reference/html/using-boot-using-springbootapplication-annotation.html#using-boot-using-springbootapplication-annotation[@SpringBootApplication] enables Spring Boot’s auto-configuration mechanism, enables `@Component` scan on the package where the application is located, and allows the application to register extra beans in the context or import additional configuration classes.
```

kroegerk commented on guides/micronaut-spring-boot/micronaut-spring-boot.adoc:
```diff
@@ -0,0 +1,169 @@
+
+=== Configuration
+
+Add some configuration which will populate this `GreetingConfiguration`:
```
```suggestion
Add a configuration to populate the `GreetingConfiguration`:
```

kroegerk commented on guides/micronaut-spring-boot/micronaut-spring-boot.adoc:
```diff
@@ -0,0 +1,169 @@
+
+=== POJO
+
+Create a POJO which uses https://github.com/FasterXML/jackson-annotations[Jackson Annotations].
```
```suggestion
Create a POJO that uses https://github.com/FasterXML/jackson-annotations[Jackson Annotations].
```

kroegerk commented on guides/micronaut-spring-boot/micronaut-spring-boot.adoc:
```diff
@@ -0,0 +1,169 @@
+
+=== @RestController
+
+Micronaut Spring only supports `@RestController` semantics. A `@RestController` request handling method serializes return objects into an HTTP Response. Create a controller which uses several Spring annotations.
```
```suggestion
Micronaut Spring only supports `@RestController` semantics. A `@RestController` request handling method serializes return objects into an HTTP response. Create a controller that uses the following Spring annotations.
```

kroegerk commented on guides/micronaut-spring-boot/micronaut-spring-boot.adoc:
```diff
@@ -0,0 +1,169 @@
+
+=== @Scheduled
+
+You can use `@Scheduled` annotation to schedule a re-occurring task. Create a Job:
```
```suggestion
You can use the `@Scheduled` annotation to schedule a recurring task. Create a Job:
```

kroegerk commented on guides/micronaut-spring-boot/micronaut-spring-boot.adoc:
```diff
@@ -0,0 +1,169 @@
+callout:constructor-di[number=1,arg0=GreetingService]
+callout:spring-ann-scheduled[3]
+
+=== Server side HTML Rendering
```
```suggestion
=== Server-side HTML Rendering
```

kroegerk commented on guides/micronaut-spring-boot/micronaut-spring-boot.adoc:
```diff
@@ -0,0 +1,169 @@
+
+==== Controller Test
+
+Add a test which shows, you can use `org.springframework.ui.Model` and render server-side HTML.
```
```suggestion
Add a test that proves you can use `org.springframework.ui.Model` and render server-side HTML.
```

kroegerk commented on guides/micronaut-spring-boot/micronaut-spring-boot.adoc:
```diff
@@ -0,0 +1,169 @@
+
+common:runapp.adoc[]
+
+You can execute the API:
```
```suggestion
Execute the API:
```

kroegerk commented on src/docs/common/callouts/callout-spring-ann-scheduled.adoc:
```diff
@@ -0,0 +1 @@
+https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/scheduling/annotation/Scheduled.html[org.springframework.scheduling.annotation.Scheduled] maps to https://docs.micronaut.io/latest/api/io/micronaut/scheduling/annotation/Scheduled.html[io.micronaut.scheduling.annotation.Scheduled] at compilation time. You can use `@Scheduled` annotation to schedule a re-occurring ...
```
```suggestion
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/scheduling/annotation/Scheduled.html[org.springframework.scheduling.annotation.Scheduled] maps to https://docs.micronaut.io/latest/api/io/micronaut/scheduling/annotation/Scheduled.html[io.micronaut.scheduling.annotation.Scheduled] at compilation time. You can use `@Scheduled` annotation to schedule a recurring task.
```

kroegerk commented on src/docs/common/callouts/callout-spring-ann-service.adoc:
```diff
@@ -0,0 +1 @@
+https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/stereotype/Service.html[org.springframework.stereotype.Service] maps to https://docs.micronaut.io/latest/api/io/micronaut/context/annotation/Bean.html[io.micronaut.context.annotation.Bean] at compilation time. A Spring's `@Service` indicates that they hold business logic.
```
I'm not sure which of these is more accurate, but it should be one of these:

```suggestion
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/stereotype/Service.html[org.springframework.stereotype.Service] maps to https://docs.micronaut.io/latest/api/io/micronaut/context/annotation/Bean.html[io.micronaut.context.annotation.Bean] at compilation time. Spring's `@Service` indicates that they hold business logic.
```
OR

```suggestion
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/stereotype/Service.html[org.springframework.stereotype.Service] maps to https://docs.micronaut.io/latest/api/io/micronaut/context/annotation/Bean.html[io.micronaut.context.annotation.Bean] at compilation time. A Spring `@Service` indicates that they hold business logic.
```

Also, I'm not sure what "they" refers to in that last sentence. Should that be clarified?

kroegerk commented on src/docs/common/callouts/callout-spring-response-entity.adoc:
```diff
@@ -0,0 +1 @@
+Special support is provided by the `micronaut-spring-web` dependency to allowing returning a https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/http/ResponseEntity.html[ResponseEntity].
```
```suggestion
Special support is provided by the `micronaut-spring-web` dependency to allow the application to return a https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/http/ResponseEntity.html[ResponseEntity].
```

sdelamo committed:
Update guides/micronaut-spring-boot/micronaut-spring-boot.adoc

Co-authored-by: Kristin Kroeger <33693760+kroegerk@users.noreply.github.com>

sdelamo committed:
Update guides/micronaut-spring-boot/micronaut-spring-boot.adoc

Co-authored-by: Kristin Kroeger <33693760+kroegerk@users.noreply.github.com>

sdelamo committed:
Update guides/micronaut-spring-boot/micronaut-spring-boot.adoc

Co-authored-by: Kristin Kroeger <33693760+kroegerk@users.noreply.github.com>

sdelamo committed:
Update guides/micronaut-spring-boot/micronaut-spring-boot.adoc

Co-authored-by: Kristin Kroeger <33693760+kroegerk@users.noreply.github.com>

sdelamo committed:
Update guides/micronaut-spring-boot/micronaut-spring-boot.adoc

Co-authored-by: Kristin Kroeger <33693760+kroegerk@users.noreply.github.com>

sdelamo committed:
Update guides/micronaut-spring-boot/micronaut-spring-boot.adoc

Co-authored-by: Kristin Kroeger <33693760+kroegerk@users.noreply.github.com>

sdelamo committed:
Update src/docs/common/callouts/callout-spring-ann-scheduled.adoc

Co-authored-by: Kristin Kroeger <33693760+kroegerk@users.noreply.github.com>

sdelamo committed:
Update src/docs/common/callouts/callout-spring-response-entity.adoc

Co-authored-by: Kristin Kroeger <33693760+kroegerk@users.noreply.github.com>

sdelamo committed:
Update src/docs/common/callouts/callout-spring-ann-service.adoc

Co-authored-by: Kristin Kroeger <33693760+kroegerk@users.noreply.github.com>
